common.ts

```diff
@@ -10,28 +10,42 @@ export class Cursor {
 export type Native<C extends Codec> = C extends Codec<infer N> ? N : never;
 
 /** A means of encoding and decoding a type `T` to and from its SCALE byte representation */
-export class Codec<T = any> {
+export abstract class Codec<T = any> {
+  /** The minimum size of values encoded with this codec */
+  abstract _minSize: number;
   /**
-   * @param _s Accepts the decoded value and returns its to-be-encoded size
-   * @param _e Accepts the cursor and value and writes its bytes into the cursor's byte array
-   * @param _d Decodes the currently-focused bytes into `T` and moves the cursor as is appropriate
+   * Accepts the decoded value and returns the size minus minSize.
+   * May be overridden by subclasses.
    */
-  constructor(
-    readonly _s: (value: T) => number,
-    readonly _e: (
-      cursor: Cursor,
-      value: T,
-    ) => void,
-    readonly _d: (cursor: Cursor) => T,
-  ) {}
+  _dynSize(value: T): number {
+    value;
```

Comment:
Why this line?

Reply:
To avoid an unused variable warning

Reply:
Does prefixing with an underscore not suppress the warning?